File: socha-sdk/src/server-api/sc/protocol/responses/GamePausedEvent.java
package sc.protocol.responses;

import com.thoughtworks.xstream.annotations.XStreamAlias;

import sc.framework.plugins.Player;

@XStreamAlias(value = "paused")
public class GamePausedEvent implements ProtocolMessage{
  
  private final Player nextPlayer;

  /** might be needed by XStream */
  public GamePausedEvent() {
    nextPlayer = null;
  }

  public GamePausedEvent(Player nextPlayer) {
    this.nextPlayer = nextPlayer;
  }

  public Player getNextPlayer() {
    return this.nextPlayer;
  }

}

File: socha-sdk/src/server-api/sc/protocol/responses/GamePausedEvent.kt
package sc.protocol.responses

import com.thoughtworks.xstream.annotations.XStreamAlias
import com.thoughtworks.xstream.annotations.XStreamAsAttribute

import sc.framework.plugins.Player

/**
 * Indicates that the game has been paused.
 *
 * @param nextPlayer the Player who will be next
 */
@XStreamAlias(value = "paused")
data class GamePausedEvent(
        @XStreamAsAttribute
        val nextPlayer: Player
): ProtocolMessage
